### ISSUE TITLE
docs for iterator: repeat / init_epoch

### ISSUE THREAD

**Issue by marikgoldstein** [pytorch/text#136]
I think people tend to get confused that when a iterator is created with repeat set to true, it's up to them to break out of the for-loop when they are ready (the iterator never stops). It seems like when people set repeat to true, they might think "yes I want to train for multiple epochs without getting the StopIterator warning".

It seems like the two modes of iteration that this library encourages are:

1. have repeat set to false, control epochs in an outer loop, call iter.init_epoch() at the start of every epoch
2. have repeat set to true, use a counter and an if statement to conditionally break out of the loop when ready

Is this roughly correct?

Happy to submit a PR to clarify this in the docs of iterator.py

**Comment by jekbradbury** [comment]
Yes, that's an accurate assessment of the current state of the library; things are the way they are because the iterator classes evolved from ones which assumed training-loop infrastructure similar to [Chainer](github.com/chainer/chainer)'s Trainer and Updater classes.
Do you have a preference between us leaving these behaviors as they are and documenting them, versus making them a little more intuitive?

**Comment by marikgoldstein** [comment]
Actually the usage pattern seems fine, maybe it's just a matter of documentation.

On the other hand, the torch dataloaders don't require you to do this.

Maybe it's best to just add documentation for now. Hope it's okay to submit some documentation !

**Comment by yasufumy** [comment]
@jekbradbury @marikgoldstein 
In my opnion, removing a repeat option is better. This argument is confusing when using torchnet.engine.Engine (I guess this is an official trainer) with Iterator, because this option should be always set to False even if iterator is used for training. According [this line](https://github.com/pytorch/tnt/blob/master/torchnet/engine/engine.py#L23), Iterator class is expected to raise StopIteration once an epoch loop is finished.
So my suggestion is changing the Iterator class to raise StopIteration after every epoch and removing a repeat option.
I think updating the documentation may help some aspect but that will not be a real solution.
What do you guys think?
